File: 플라잉센터자동화/app/display.py
"""Display/label helper functions for UI rendering."""
from __future__ import annotations

from typing import Optional


def display_payment_method(payment_method: Optional[str]) -> str:
    if not payment_method:
        return "-"
    mapping = {
        "PAY_QR": "QR결제",
        "CASH": "현금",
    }
    return mapping.get(payment_method, payment_method)


def display_flying_pass_tier(tier: str) -> str:
    from app.services.flying_pass import FLYING_PASS_TIER_LABELS, normalize_flying_pass_tier

    normalized_tier = normalize_flying_pass_tier(tier)
    return FLYING_PASS_TIER_LABELS.get(normalized_tier, normalized_tier)


def display_lost_found_status(status_value: str) -> str:
    mapping = {
        "STORED": "보관중",
        "RETURNED": "인계완료",
        "DISPOSED": "폐기",
    }
    return mapping.get(status_value, status_value)


def display_note_category(category: str) -> str:
    mapping = {
        "NOTICE": "안내사항",
        "HANDOVER": "인수인계",
    }
    return mapping.get(category, category)


def display_cash_closing_type(closing_type: str) -> str:
    mapping = {
        "MORNING_HANDOVER": "오전 인수인계",
        "FINAL_CLOSE": "최종 마감",
    }
    return mapping.get(closing_type, closing_type)


def display_cash_closing_status(status_value: str) -> str:
    mapping = {
        "DRAFT": "작성중",
        "SUBMITTED": "제출됨",
        "LOCKED": "확인완료(잠금)",
    }
    return mapping.get(status_value, status_value)


def cash_closing_status_pill_class(status_value: str) -> str:
    mapping = {
        "DRAFT": "status-payment_pending",
        "SUBMITTED": "status-picked_up",
        "LOCKED": "status-paid",
    }
    return mapping.get(status_value, "status-payment_pending")


def display_cash_closing_audit_action(action: str) -> str:
    mapping = {
        "CREATE": "등록",
        "UPDATE": "수정",
        "ADMIN_UNLOCK_UPDATE": "잠금해제수정",
        "SUBMIT": "제출",
        "VERIFY_LOCK": "확인잠금",
    }
    return mapping.get(action, action)

File: worker/src/lib/display.ts
/**
 * Korean label helpers for enum values displayed in UI.
 */

export function displayPaymentMethod(method: string | null): string {
  if (method === null) return "-";
  switch (method) {
    case "PAY_QR": return "QR결제";
    case "CASH": return "현금";
    case "MIXED": return "분할결제";
    default: return method;
  }
}

export function displayFlyingPassTier(tier: string): string {
  switch (tier) {
    case "NONE": return "미적용";
    case "BLUE": return "블루";
    case "SILVER": return "실버";
    case "GOLD": return "골드";
    case "PLATINUM": return "플래티넘";
    case "BLACK": return "블랙";
    default: return tier;
  }
}

export function displayOrderStatus(status: string): string {
  switch (status) {
    case "PAYMENT_PENDING": return "결제대기";
    case "PAID": return "결제완료";
    case "PICKED_UP": return "수령완료";
    case "CANCELLED": return "취소";
    default: return status;
  }
}

export function displayLostFoundStatus(status: string): string {
  switch (status) {
    case "UNCLAIMED": return "보관중";
    case "CLAIMED": return "인계완료";
    case "DISPOSED": return "폐기";
    default: return status;
  }
}

export function displayNoteCategory(category: string): string {
  switch (category) {
    case "NOTICE": return "안내사항";
    case "HANDOVER": return "인수인계";
    default: return category;
  }
}

export function displayCashClosingType(type: string): string {
  switch (type) {
    case "MORNING_HANDOVER": return "오전 인수인계";
    case "FINAL_CLOSE": return "최종 마감";
    default: return type;
  }
}

export function displayCashClosingStatus(status: string): string {
  switch (status) {
    case "DRAFT": return "작성중";
    case "SUBMITTED": return "제출됨";
    case "LOCKED": return "확인완료(잠금)";
    default: return status;
  }
}
